src/utils/breachResolution.js

```diff
@@ -141,32 +147,72 @@ function getResolutionRecsPerBreach (dataTypes, args, options = {}) {
   const filteredBreachRecs = {}
 
   // filter breachResolutionDataTypes based on relevant data types passed in
-  for (const [key, value] of Object.entries(breachResolutionDataTypes)) {
+  for (const resolution of Object.entries(breachResolutionDataTypes)) {
+    const [key, value] = resolution
     if (
       dataTypes.includes(key) &&
-      // Hide the security question or password resolution if we can't link to the breached site:
-      (![BreachDataTypes.Passwords, BreachDataTypes.SecurityQuestions].includes(key) || !options.hideBreachLink) &&
       // Hide resolutions that apply to other countries than the user's:
       (!options.countryCode || !Array.isArray(value.applicableCountryCodes) || value.applicableCountryCodes.includes(options.countryCode.toLowerCase()))
     ) {
-      filteredBreachRecs[key] = getRecommendationFromResolution(value, args)
+      filteredBreachRecs[key] = getRecommendationFromResolution(resolution, args)
     }
   }
 
   // If we did not have any recommendations, add a generic recommendation:
   if (Object.keys(filteredBreachRecs).length === 0) {
-    filteredBreachRecs[BreachDataTypes.General] = getRecommendationFromResolution(breachResolutionDataTypes[BreachDataTypes.General], args)
+    const resolutionTypeGeneral = BreachDataTypes.General
+    filteredBreachRecs[resolutionTypeGeneral] = getRecommendationFromResolution(
+      [
+        resolutionTypeGeneral,
+        breachResolutionDataTypes[resolutionTypeGeneral]
+      ],
+      args
+    )
   }
 
   // loop through the breach recs
   return filteredBreachRecs
 }
 
+/**
+ * Get the fluent string for the body
+ *
+ * @param {string} body for the fluent body string
+ * @param {object} args
+ * @returns {string} body string
+ */
+function getBodyMessage (body, args) {
+  const { resolutionType, stringArgs } = args
+
+  switch (resolutionType) {
+    case BreachDataTypes.Passwords:
+    case BreachDataTypes.SecurityQuestions: {
```

Comment:
Personal preference, so feel free to ignore, but I'd keep the code simple here and just replace `<breached-company-link>` in every string, rather than just the passwords and security questions resolutions.

Reply:
No hard preference from my side: I thought being a bit more specific here was OK since we were cautious about how and where we link out to. Less complex is also a good thing, though: [d574bee](https://github.com/mozilla/blurts-server/pull/2961/commits/d574bee881fa1d55364509f8e1735a34596204dd).